qiskit/algorithms/evolvers/pvqd/pvqd.py

```diff
@@ -0,0 +1,364 @@
+        """Set the quantum instance and circuit sampler."""
+        if quantum_instance is not None:
+            if not isinstance(quantum_instance, QuantumInstance):
+                quantum_instance = QuantumInstance(quantum_instance)
+            self._sampler = CircuitSampler(quantum_instance)
+
+        self._quantum_instance = quantum_instance
+
+    def step(
+        self, hamiltonian: OperatorBase, theta: np.ndarray, dt: float, initial_guess: np.ndarray
+    ) -> Tuple[np.ndarray, float]:
+        """Perform a single time step.
+
+        Args:
+            hamiltonian: The Hamiltonian under which to evolve.
+            theta: The current parameters.
+            dt: The time step.
+            initial_guess: The initial guess for the classical optimization of the
+                fidelity between the next variational state and the Trotter-evolved last state.
+                If None, this is set to a random vector with elements in the interval
+                :math:`[-0.01, 0.01]`.
+
+        Returns:
+            A tuple consisting of the next parameters and the fidelity of the optimization.
+        """
+        loss, gradient = self.get_loss(hamiltonian, dt, theta)
+
+        if initial_guess is None:
+            initial_guess = np.random.random(self.initial_parameters.size) * 0.01
+
+        if isinstance(self.optimizer, Optimizer):
+            optimizer_result = self.optimizer.minimize(loss, initial_guess, gradient)
+        else:
+            optimizer_result = self.optimizer(loss, initial_guess, gradient)
+
+        # clip the fidelity to [0, 1]
+        fidelity = np.clip(1 - optimizer_result.fun, 0, 1)
+
+        return theta + optimizer_result.x, fidelity
+
+    def get_loss(
+        self,
+        hamiltonian: OperatorBase,
+        dt: float,
+        current_parameters: np.ndarray,
+    ) -> Tuple[Callable[[np.ndarray], float], Optional[Callable[[np.ndarray], np.ndarray]]]:
+
+        """Get a function to evaluate the infidelity between Trotter step and ansatz.
+
+        Args:
+            hamiltonian: The Hamiltonian under which to evolve.
+            dt: The time step.
+            current_parameters: The current parameters.
+
+        Returns:
+            A callable to evaluate the infidelity and, if gradients are supported and required,
+                a second callable to evaluate the gradient of the infidelity.
+        """
+        self._validate_setup()
+
+        # use Trotterization to evolve the current state
+        trotterized = self.ansatz.bind_parameters(current_parameters)
+        trotterized.append(
+            PauliEvolutionGate(hamiltonian, time=dt, synthesis=self.evolution), self.ansatz.qubits
+        )
+
+        # define the overlap of the Trotterized state and the ansatz
+        x = ParameterVector("w", self.ansatz.num_parameters)
+        shifted = self.ansatz.assign_parameters(current_parameters + x)
+        overlap = StateFn(trotterized).adjoint() @ StateFn(shifted)
```

Comment:
I may increase entropy with this comment but do we maybe want to keep more flexibility for the user to enable the employment of alternative overlap measures? Maybe we don't want to do it right now. I just wanted to raise to point to create awareness :)

Reply:
Good point, I would postpone this to once we have fidelities as objects (see #8303) then we could allow a `Fidelity` as  input 🙂